ts/calculator.ts

```diff
@@ -0,0 +1,476 @@
+const f_block = document.getElementById("f-block")!;
+const func_arrow = document.getElementById("func_arrow")!;
+const trig_arrow = document.getElementById("trig_arrow")!;
+const other_funcs = document.getElementById("other-funcs")!;
+
+other_funcs.addEventListener("click", () => {
+    changeFuncs();
+})
+const FACTORIAL = "FACTORIAL";
+const OPERATORS = ["+", "-", "*", "/"];
+interface Exp_data {
+    operation: (string | number | boolean)[],
+    formula: (string | number | boolean)[],
+}
+let data: Exp_data = {
+    operation: [],
+    formula: [],
+};
+let ans: number | string = 0;
+let result: number | string;
+let btn: string;
+const input_element = document.querySelector(".btn-wrap")!;
+const output_operation_element = document.querySelector("#display") as HTMLInputElement;
+let RADIAN = false;
+let memory_value = 0;
+const deg_btn = document.getElementById("deg")!;
+
+trig_arrow.addEventListener("click", (event) => {
+    event.stopPropagation();
+});
+
+trig_block.addEventListener("click", (event) => {
+    event.preventDefault();
+    if (trig_container.style.display === "block") {
+        trig_container.style.display = "none";
+        trig_arrow.style.transform = "rotate(90deg)";
+    } else {
+        trig_container.style.display = "block";
+        trig_arrow.style.transform = "rotate(270deg)";
+    }
+});
+
+f_block.addEventListener("click", () => {
+    if (func_container.style.display === "block") {
+        func_container.style.display = "none";
+        func_arrow.style.transform = "rotate(90deg)";
+    } else {
+        func_container.style.display = "block";
+        func_arrow.style.transform = "rotate(270deg)";
+    }
+});
+
+function angleToggle() {
+    if (RADIAN) {
+        deg_btn.innerHTML = 'RAD';
+    } else {
+        deg_btn.innerHTML = 'DEG';
+    }
+    return false;
+}
+
+function negate(num: number) {
+    num = -num;
+    return num;
+}
+
+function toExponent(number: number) {
+    let expval = number.toExponential();
+    return expval;
+}
+
+function changeFuncs() {
+    let square_cube = document.getElementById("sq-cube")!.innerHTML;
+    let square_cube_root = document.getElementById("sq-cube-root")!.innerHTML;
+    if (square_cube === "x²") {
+        document.getElementById("sq-cube")!.innerHTML = "x​³";
+    } else {
+        document.getElementById("sq-cube")!.innerHTML = "x²";
+    }
+    if (square_cube_root === "√x") {
+        document.getElementById("sq-cube-root")!.innerHTML = "∛x";
+    } else {
+        document.getElementById("sq-cube-root")!.innerHTML = "√x";
+    }
+}
+
+input_element.addEventListener("click", (event) => {
+    const target_btn = event.target as Element;
+    btn = target_btn.innerHTML;
+    for (let button of calculator_buttons) {
+        if (button.symbol == target_btn.innerHTML) {
+            calculator(button);
+            memoryFunction(btn);
+            break;
+        }
+    }
+});
+
+function memoryFunction(btn: string) {
```

Comment:
Instead of using `string`, create a type alias for all possible values of string.

Reply:
value of btn will always be string, therefore i added string

Reply:
type of btn will always be string,therefore i added string type

Reply:
In this case, string type can be reduced to union of `MC, MR, M+` , and so on.
No need to use `string` super set. This ensures strict type check at compile time.